tests/unit/test_util.cpp

```diff
@@ -78,3 +78,37 @@ TEST_F(UtilTest, testFromHex)
 
     ASSERT_THROW(result = mococrw::utility::fromHex("  deadbeef"), mococrw::MoCOCrWException);
 }
+
+TEST_F(UtilTest, stringCleanse)
+{
+    std::string secret{"1234"};
+    mococrw::utility::stringCleanse(secret);
+    ASSERT_THAT(secret, Eq(std::string()));
+}
+
+TEST_F(UtilTest, stringCleanseCheckMemory)
+{
+    std::string secret{"1234"};
+    auto secretSize = secret.size();
+    std::string zeroedOutString(secretSize, '\0');
+    mococrw::utility::stringCleanse(secret);
+    ASSERT_GE(secret.capacity(), secretSize);
+    ASSERT_THAT(0, memcmp(&secret[0], &zeroedOutString[0], secretSize));
```

Comment:
This test relies on undefined behavior.
The reference actually says that 
> If pos > size(), the behavior is undefined.

It does not refer to capacity. So while I guess this mostly works, therei s no strong guarantee for this. If you want to test if the memory was nulled, it needs to happen before the size is reset. 
I also don't think that an implementation needs to keep the space around. reset() may free the buffer.
So, while I appreciate the effort here, I don't think that this is the best way to test this.
If we really want to test for this, we should get a pointer to the heap fragment that contains the string before cleaning and then checking lateron that nothing from the original data is left there.

Reply:
Done